File: frontend/src/types/vacancy.ts
export type OptionalApplicationField = 'applicantPhone' | 'file' | 'certificates'

export interface VacancyInput {
  title: string
  description: string
  requiredFields: OptionalApplicationField[]
}

export interface Vacancy {
  id: string
  title: string
  description: string
  isOpen: boolean
  createdAt: string
  filledByApplicationId: string | null
  requiredFields: OptionalApplicationField[]
}

File: backend/src/main/java/taller/multimedia/backend/model/vacancy/Vacancy.java
package taller.multimedia.backend.model.vacancy;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "vacancies")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Vacancy {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "is_open", nullable = false)
    private boolean open = true;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "filled_by_application_id")
    private UUID filledByApplicationId;

    // Lista separada por comas de OptionalApplicationField (applicantPhone, file, certificates)
    @Column(name = "required_fields")
    private String requiredFields;
}
